docs/reference/ilm/apis/slm-api.asciidoc

```diff
@@ -21,36 +21,166 @@ by a different enable setting. To disable SLM's functionality, set the cluster
 
-==== Request Parameters
 
-include::{docdir}/rest-api/common-parms.asciidoc[tag=timeoutparms]
+[[slm-api-put-prereqs]]
+==== {api-prereq-title}
 
-==== Authorization
+If you use {es} {security-features},
+you must have:
 
-You must have the `manage_slm` cluster privilege to use this API. You must also
-have the `manage` index privilege on all indices being managed by `policy`. All
-operations executed by {slm} for a policy are executed as the user that put the
-latest version of a policy. For more information, see
-<<security-privileges>>.
+* `manage_slm` <<privileges-list-cluster,cluster privileges>>
+* `manage` <<privileges-list-indices,index privileges>> for any included indices
 
-==== Example
+{slm-cap} operations are executed
+as the user that last created or updated the policy.
+
+For more information,
+see <<security-privileges>>.
+
+
+[[slm-api-put-desc]]
+==== {api-description-title}
+
+Use the put snapshot lifecycle policy API
+to create or update a snapshot lifecycle policy.
+
+If the policy already exists,
+this request increments the policy's version.
+Only the latest version of the policy is stored.
+
+
+
+[[slm-api-put-path-params]]
+==== {api-path-parms-title}
 
-The following creates a snapshot lifecycle policy with an id of
-`daily-snapshots`:
+`<snapshot-lifecycle-policy-id>`::
+(Required, string)
+ID for the snapshot lifecycle policy
+you want to create or update.
+
+
+[[slm-api-put-query-params]]
+==== {api-query-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=timeoutparms]
+
+
+[[slm-api-put-request-body]]
+==== {api-request-body-title}
+
+`schedule`::
+(Required, <<schedule-cron,Cron scheduler configuration>>)
+Periodic or absolute schedule
+at which the policy creates snapshots
+and deletes expired snapshots.
++
+Schedule changes to existing policies
+are applied immediately.
+
+`name`::
++
+--
+(Required, string)
+Name automatically assigned to each snapshot
+created by the policy.
+
+This value supports the same <<date-math-index-names,date math>>
+supported in index names.
+
+To prevent conflicting snapshot names,
+a UUID is automatically appended to each snapshot name.
+--
+
+`repository`::
++
+--
+(Required, string)
+Repository used to store snapshots
+created by this policy.
+
+This repository must exist prior to the policy's creation.
+You can create a respository
+using the <<modules-snapshots,snapshot repository API>>.
+--
+
+`config`::
++
+--
+(Required, object)
+Configuration for each snapshot
+created by the policy.
+
+Parameters include:
+
+`indices`::
+(Optional, array of strings)
+Array of index names or wildcard pattern of index names
+included in snapshots.
+
+`ignore_unavailable`::
+(Optional, boolean)
+If `true`,
+missing indices do *not* cause snapshot creation to fail
+and return an error.
+Defaults to `false`.
+
+`include_global_state`::
+(Optional, boolean)
+If `true`,
+cluster states are included in snapshots.
+Defaults to `false`.
+--
+
+`retention`::
```

Comment:
This whole section cannot be backported to 7.4, since retention does not exist in that version